### PR TITLE
Sound fix for when placing block against cb

### PR DESCRIPTION
**Pull request by minustenchan**
Fixes BlockPlaceEvent being called even though block cannot be placeable there, thus playing a required.wood/stone.place sound

should fix pretty much every issue there is that I know of for placing against custom block

Tested in 1.20.4 and 1.19.4

### CODE REVIEW

**Comment by minustenchan** [comment]
actually I thought the arm swing animation bug was fixed when placing custom block against replaceable block, but turns out it's not, got zero clue why it just doesn't swing the arm when placing custom block against replaceable

File: core/src/main/java/io/th0rgal/oraxen/mechanics/provided/gameplay/noteblock/NoteBlockMechanicListener.java [hunk 1]
```diff
@@ -16,6 +16,7 @@
 import io.th0rgal.oraxen.utils.breaker.BreakerSystem;
 import io.th0rgal.oraxen.utils.breaker.HardnessModifier;
 import io.th0rgal.protectionlib.ProtectionLib;
+import net.minecraft.world.InteractionResult;
```
Comment (Boy0000) [inline core/src/main/java/io/th0rgal/oraxen/mechanics/provided/gameplay/noteblock/NoteBlockMechanicListener.java:19]: This introduces an NMS-class to non-nms packages
In theory this should be fine aslong as this class exists in all versions
If there is an API equivalent I would like to see it used instead

**Review by Boy0000 — CHANGES_REQUESTED** [review]
As a sidenote I think this might be better aimed in general at 2.0
2.0 has tons of changes and as this isnt vital, would rather see it implemented there

Easier than rebasing these changes onto the very changed 2.0 codebase

File: core/src/main/java/io/th0rgal/oraxen/mechanics/provided/gameplay/noteblock/NoteBlockMechanicListener.java (hunk 1 above)
Comment (syveriea) [reply core/src/main/java/io/th0rgal/oraxen/mechanics/provided/gameplay/noteblock/NoteBlockMechanicListener.java:19]: pretty sure it exists from all the way to 1.18 to 1.20.4

**Comment by minustenchan** [comment]
> As a sidenote I think this might be better aimed in general at 2.0 2.0 has tons of changes and as this isnt vital, would rather see it implemented there
> 
> Easier than rebasing these changes onto the very changed 2.0 codebase

will open a PR for 2.0 👍

**Pull request closed without merging**